test/generator/service_test.exs

```diff
@@ -296,43 +308,76 @@ defmodule Thrift.Generator.ServiceTest do
     assert {:tags_with_options, 3}              in defined_functions
   end
 
-
   # connection tests
 
   thrift_test "clients can be closed", ctx do
     :ok = Client.close(ctx.client)
   end
 
-  thrift_test "clients retry on a closed server", ctx do
-    ref = Process.monitor(ctx.server)
-    :thrift_socket_server.stop(ctx.server)
+  thrift_test "clients don't retry by default", ctx do
+    Process.flag(:trap_exit, true)
+    stop_server(ctx.server)
+    {:ok, new_server} = start_server(ctx.port, 20)
 
-    assert_receive {:DOWN, ^ref, _, _, _}
+    {:ok, client} = Client.start_link("127.0.0.1", ctx.port,[])
+
+    :timer.sleep(50) # sleep beyond the server's recv_timeout
+    assert catch_exit(Client.friend_ids_of(client, 1234))
+
+    on_exit fn ->
+      stop_server(new_server)
+    end
+  end
+
+  thrift_test "clients retry when making an RPC on a closed server when retry is true", ctx do
+    stop_server(ctx.server)
+    {:ok, server} = start_server(ctx.port, 20)
+    {:ok, client} = Client.start_link("127.0.0.1", ctx.port, [tcp_opts: [retry: true]])
+    :timer.sleep(50)
```

Comment:
This test does:

1. stop server
2. start server
3. start client

Why stop/start the server? The point should be that the server is not running when the client starts (and tries to connect). I think it's supposed to be like the next test which does:

1. stop server
2. start client
3. start server